bson/mgocompat/registry.go

```diff
@@ -78,7 +78,8 @@ func newRegistryBuilder() *bsoncodec.RegistryBuilder {
 		RegisterTypeMapEntry(bsontype.DateTime, tTime).
 		RegisterTypeMapEntry(bsontype.Array, tInterfaceSlice).
 		RegisterHookEncoder(tGetter, bsoncodec.ValueEncoderFunc(GetterEncodeValue)).
-		RegisterHookDecoder(tSetter, bsoncodec.ValueDecoderFunc(SetterDecodeValue))
+		RegisterHookDecoder(tSetter, bsoncodec.ValueDecoderFunc(SetterDecodeValue)).
+		RegisterTypeMapEntry(bsontype.EmbeddedDocument, reflect.TypeOf(primitive.M{}))
```

Comment:
You can store your reflect.TypeOf(...) stuff in the var block at the top. Also, we don't need the bsonType.Type(0) map entry anymore.

Reply:
Done.